packages/db/src/runtime/db-client.ts

```diff
@@ -81,11 +81,24 @@ function createRemoteLibSQLClient(appToken: string, remoteDbURL: URL, rawUrl: st
 		url = 'file:' + remoteDbURL.pathname.substring(1);
 	}
 
-	const client = createClient({
-		...options,
-		authToken: appToken,
-		url,
-	});
+	// this function parses the options from a `Record<string, string>`
+	// provided from the object conversion of the searchParams and properly
+	// verifies that the Config object is providing the correct types.
+	// without this, there is runtime errors due to incorrect values
+	function parseOpts(config: Record<string, string>): LibSQLConfig {
+		return {
+			... config,
+			syncInterval: config.syncInterval ? parseInt(options.syncInterval) : undefined,
+			readYourWrites: config.readYourWrites ? !!config.readYourWrites : undefined,
+			offline: config.offline ? !!config.offline : undefined,
+			tls: config.tls ? !!config.tls : undefined,
+			concurrency: config.concurrency ? parseInt(options.concurrency) : undefined,
+			authToken: appToken,
```

Comment:
Passing `?readYourWrites` would result in the flag not being passed since an empty string is a falsy value. For the boolean values I think the following would be better:

```suggestion
			syncInterval: config.syncInterval ? parseInt(options.syncInterval) : undefined,
			readYourWrites: 'readYourWrites' in config ? config.readYourWrites !== 'false' : undefined,
			offline: 'offline' in config ? config.offline !== 'false' : undefined,
			tls: 'tls' in config ? config.tls !== 'false' : undefined,
			concurrency: config.concurrency ? parseInt(options.concurrency) : undefined,
			authToken: appToken,
```

This way not passing a boolean value uses the default value, passing `?readYourWrites` enables the flag explicitly, and passing `?readYourWrites=false` disables the flag explicitly.